### DISCUSSION TITLE
Display spectrogram for timing in editor

### DISCUSSION THREAD

**Discussion by ghost** [ppy/osu#23392] [Ideas] [answered]
When timing, it's hard to tell where the peak is here just through these graphics (the intensity/brightness is at a lower resolution too), and just listening alone won't help.

![osu!lazer editor, peaks](https://user-images.githubusercontent.com/90960821/235923510-d7b14a3d-6f6f-432b-b598-1d29c4c28617.png)

Things like this are better off displayed in a spectrogram, where you can tell where the impacts are. Below is a screenshot of BASS's spectrum example in C:

![image](https://user-images.githubusercontent.com/90960821/235924484-923c41a7-38ea-4cb3-bad7-54e1b0564c98.png)

While this isn't a total disaster of an issue in lazer, it's a bit bothersome to have to open an external program to find timings, so this is a suggestion to have something like this in lazer, both in the main timeline and the little preview window next to the metronome.

**Comment by peppy** [comment, marked as answer]
Have you tried using the timing interface in lazer? You shouldn't need the spectogram if you use it.

I'm against adding this from an accuracy and simplicity angle. Misunderstanding how attack works in sound can result in very incorrect timing using such imagery, which is why I want to keep timing as aural as possible.

**Reply by ghost** [reply]
> Have you tried using the timing interface in lazer?

Yes, it's not all that great IMO, plus I am getting this bug https://github.com/ppy/osu/issues/23393 so it's even more confusing when I try to time a beatmap purely through sound.
I should also open another issue on how https://github.com/ppy/osu-resources/blob/master/osu.Game.Resources/Samples/UI/metronome-tick.wav has a bit of silence at the beginning.

**Reply by peppy** [reply]
Definitely worth opening an issue for the new metronome sound having ~5ms of silence/attack, which is too high for timing purposes cc/ @nekodex

**Reply by ghost** [reply]
> Definitely worth opening an issue for the new metronome sound having ~5ms of silence/attack, which is too high for timing purposes cc/ @nekodex

Opened one at https://github.com/ppy/osu/issues/23409
